net/redisclient.go

```diff
@@ -25,6 +26,11 @@ import (
 type RedisOptions struct {
 	// Addrs are the list of redis shards
 	Addrs []string
+
+	// AddrUpdater is a func that is called with a channel that
+	// will be written to an updated redis address list
+	AddrUpdater func(quit chan struct{}, ch chan<- []string)
```

Comment:
Maybe it should be just a channel of updates - if not nil then redis client subscribes to it and updates address on each receive.

Reply:
changed the signature to `AddrUpdater func() []string` to used it here in a goroutine to poll kubernetes dataclient